### ISSUE TITLE
ContactStatus can be an enum

### ISSUE THREAD

**Issue by nrempel** [masalachai/epp-client#32]
Possible values:

clientDeleteProhibited, clientHold, clientRenewProhibited, clientTransferProhibited, clientUpdateProhibited, inactive, ok, pendingCreate, pendingDelete, pendingRenew, pendingTransfer, pendingUpdate, serverDeleteProhibited, serverHold, serverRenewProhibited, serverTransferProhibited, serverUpdateProhibited

https://datatracker.ietf.org/doc/html/rfc5731#section-2.3

This might be more ergonomic for developers, what do you think?

**Comment by masalachai** [comment]
Yes, this is a good idea, along with having similar enums for things like 2-letter country codes (there has to be a crate that already provides this?).

**Comment by djc** [comment]
That definitely exists, https://crates.io/crates/celes looks like a decent option.

**Comment by djc** [comment]
I tried and I couldn't get the serde impls to line up correctly. I think we'd need https://github.com/tafia/quick-xml/pull/304 to be able to do this without adding a lot of custom (de)serialization code.

**Comment by masalachai** [comment]
Hi @djc,

I tried with an `Serialize` and `Deserialize` impl for an enum with the expected statuses in #37 , and updated the `contact_update()` and `contact_info()` tests for it. Let us know what you think.
